bluesky/callbacks/core.py

```diff
@@ -351,3 +352,46 @@ def start(self, doc):
     def _print(self, out_str):
         self._rows.append(out_str)
         self._out(out_str)
+
+
+class Retrieve(CallbackBase):
+    def __init__(self, root_map=None, handler_reg=None):
+        if root_map is None:
+            root_map = {}
+        if handler_reg is None:
+            handler_reg = {}
+        self.resources = None
+        self.handlers = None
+        self.datums = None
+        self.root_map = root_map
+        self.handler_reg = handler_reg
+
+    def start(self, doc):
+        self.resources = {}
```

Comment:
These are never cleared. Should they be?

Reply:
What do you mean they are never cleared? I guess I only wanted to deal with data from the run in question, so I cleared things out.

Reply:
If it is meant to be a long-lived instance, the `resources` can grow large.

Reply:
Hmm. Is making the dict anew with every start not enough to clear the resources? Or should this be cleared more frequently? If so, how often?

Reply:
You're right. Now I have no idea what I was thinking with that original comment.